### PR TITLE
Upgrade to typescript 4.1.3

### LINKED ISSUE #4800: Upgrade to Typescript version 4
We currently use typescript version 3. Most of our partners are on 4, and we think there are improvements around incremental build in 4 that will increase the stability of our build process.

@skylerjokiel this could impact your area? Any thoughts?

FYI @curtisman

### PR DESCRIPTION
**Pull request by curtisman**
Close #4800

### CODE REVIEW

**Review by tylerbutler — APPROVED** [review]
:shipit:

File: packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ export class OdspDriverUrlResolver implements IUrlResolver {
             }
         }
 
-        const summarizer = request.headers?.[DriverHeader.summarizingClient];
+        const summarizer = !!request.headers?.[DriverHeader.summarizingClient];
```
Comment (skylerjokiel) [inline packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts:103]: Doesn't this change `summarizer` changes from type `string | undefined` to `boolean`?

**Review by skylerjokiel — APPROVED** [review]
I left one comment 🍁

File: packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts (hunk 1 above)
Comment (curtisman) [reply packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts:103]: yep. it is used later on with a definite `boolean` type.

**Pull request merged**